src/google/protobuf/compiler/js/js_generator.cc

```diff
@@ -2533,13 +2548,23 @@ void Generator::GenerateClassFieldFromObject(
           JSFieldIndex(field), "fieldclass", SubmessageTypeRef(options, field));
     }
   } else if (field->type() == FieldDescriptor::TYPE_ENUM) {
-    printer->Print(
-        "  obj.$name$ != null && jspb.Message.setField(msg, $index$, "
-        "typeof obj.$name$ === \"string\" ? $enumprefix$$enumname$[obj.$name$] : obj.$name$);\n",
-        "name", JSObjectFieldName(options, field),
-        "index", JSFieldIndex(field),
-        "enumprefix", GetEnumPathPrefix(options, field->enum_type()),
-        "enumname", field->enum_type()->name());
+    if (!field->is_repeated()) {
+      printer->Print(
+           "  obj.$name$ != null && jspb.Message.setField(msg, $index$, "
+           "typeof obj.$name$ === \"string\" ? $enumprefix$$enumname$[obj.$name$] : obj.$name$);\n",
+           "name", JSObjectFieldName(options, field),
+           "index", JSFieldIndex(field),
+           "enumprefix", GetEnumPathPrefix(options, field->enum_type()),
+           "enumname", field->enum_type()->name());
+    } else {
+      printer->Print(
+           "  obj.$name$ != null && jspb.Message.setField(msg, $index$, "
+           "(obj.$name$ && obj.$name$.map(v => $enumprefix$$enumname$[obj.$name$]).filter(v => v !== undefined) || undefined));\n",
+           "name", JSObjectFieldName(options, field),
+           "index", JSFieldIndex(field),
+           "enumprefix", GetEnumPathPrefix(options, field->enum_type()),
+           "enumname", field->enum_type()->name());
```

Comment:
This generates:
![image](https://user-images.githubusercontent.com/7275468/129627910-617e0748-8152-4b37-a953-627d546c2acc.png)